addons/mail/static/src/components/message_list/message_list.js

```diff
@@ -293,7 +298,7 @@ class MessageList extends Component {
         if (!this.props.hasSquashCloseMessages) {
             return false;
         }
-        if (Math.abs(message.date.diff(prevMessage.date)) > 60000) {
+        if (message.date && Math.abs(message.date.diff(prevMessage.date)) > 60000) {
```

Comment:
You also need to guard for `prevMessage.date` not being set, for which it should not be squashed (no prev date but a date on current message: we need to avoid squashing to display the date of the current message).